src/test/java/org/hisp/dhis/integration/rapidpro/route/FetchScheduledTrackerEventsRouteBuilderFunctionalTestCase.java

```diff
@@ -422,11 +423,38 @@ public void testCreateRapidProContactWhenContactAlreadyExists()
             .body( "results.size()", equalTo( 1 ) );
     }
 
+    @Test
+    public void testFetchAndProcessEvents()
+        throws
+        Exception
+    {
+        String phoneNumber = "12345678";
+        String givenName = "John";
+        String enrollmentId = Environment.createDhis2TrackedEntityWithEnrollment( Environment.ORG_UNIT_ID, phoneNumber,
+            "ID-1", givenName, List.of( "ZP5HZ87wzc0" ) );
+        AdviceWith.adviceWith( camelContext, "Fetch And Process Tracker Events",
+            r -> r.weaveByToUri( "jms:queue:events?exchangePattern=InOnly" ).after().to( "mock:spy" ) );
+        MockEndpoint spyEndpoint = camelContext.getEndpoint( "mock:spy", MockEndpoint.class );
+        spyEndpoint.setExpectedCount( 1 );
+        camelContext.start();
+        producerTemplate.sendBody( "direct:fetchAndProcessEvents", ExchangePattern.InOut, null );
```

Comment:
I've been a bit back and forth with the naming of the routes in this route builder to be honest. On one hand the main functionality is to fetch scheduled events and prepare them for further processing, but this gets muddled by the necessity of fetching additional attributes from the enrollment and having to create RapidPro contacts. I have been considering separating the event processing from the contact creation,  but this will again lead to more complexities in the `FlowStarter` routes which will be implemented next. 

For now I will rename the route to `direct:createRapidProContacts`.

Reply:
I am working on a refactor of this route builder compared to the original design, as to make it a bit more clear and separate concerns between DHIS2 and RapidPro. It then also gets easier to name the routes more precisely. Once the event data has been added to the queue, it is assumed that the corresponding Program Stage Flow in RapidPro can be triggered. 

A bit more clearly:

**Initial Design in the Bangalore use-case, all in one route builder:**
- Fetch events from DHIS2
- Fetch additional enrollment attributes for these events
- Create RapidPro contacts for these enrollments
- Transform the data
- Trigger RapidPro flow. add flow triggers to retry database if the contact is already in an active flow.
- After the flow trigger, update the event status in DHIS2 from "SCHEDULED" to "ACTIVE"

**After Refactoring:**
Split the process into two distinct parts to clearly separate concerns between DHIS2 and RapidPro.

**Part 1 - Fetch Program Stage Events Route Builder (DHIS2-related tasks):**
- Fetch events from DHIS2
- Fetch additional enrollment attributes for these events
- Transform and queue the data for the RapidPro flow trigger route
- Update the event status of the DHIS2 events from "SCHEDULED" to "ACTIVE"

**Part 2 - Trigger RapidPro Flow Route Builder (RapidPro-related tasks):**
- Dequeue event data (assuming completion of DHIS2 processing and event status update)
- Create RapidPro contacts
- Check if contact is already in a flow and handle accordingly (either trigger the flow or add to a retry queue)

So, new structure for this route (can move this into the servlet task route, as you suggested):
```
from("direct:entryPointOrchestration")
            .routeId("Fetch Events Entry Point Orchestration")
            .to( "direct:fetchDueProgramStageEvents")
            .split( simple( "${exchangeProperty.dueEvents}" ) )
                .to( "direct:retrieveEnrollmentAttributes" )
                .to( "direct:transformEvent" ) // simple datasonnet transformation to clean up the payload before queueing
                .to( "jms:queue:transformedEvent?exchangePattern=InOnly" )
                .to("direct:updateDhisProgramStageEventStatus")
```